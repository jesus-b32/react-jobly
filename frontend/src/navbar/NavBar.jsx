import React, { useState, useContext } from 'react';
import { NavLink } from 'react-router-dom';
import UserContext from '../auth/UserContext';
import {
  Navbar,
  Nav,
  NavItem,
  NavbarBrand,
  NavbarToggler,
  Collapse,
} from 'reactstrap';

function NavBar({ logout }) {
  const { currentUser } = useContext(UserContext);
  const [isOpen, setIsOpen] = useState(false);

  const toggle = () => setIsOpen(!isOpen);

  return (
    <div className='mb-4'>
      <Navbar expand={'md'} color='dark' dark>
        <NavbarBrand href='/'>React-Jobly</NavbarBrand>
        <NavbarToggler onClick={toggle} />
        <Collapse isOpen={isOpen} navbar>
          {currentUser ? (
            <Nav className='ms-auto'>
              <NavItem className='ms-4'>
                <NavLink to='/companies'>Companies</NavLink>
              </NavItem>
              <NavItem className='ms-4'>
                <NavLink to='/jobs'>Jobs</NavLink>
              </NavItem>
              <NavItem className='ms-4'>
                <NavLink to='/profile'>Profile</NavLink>
              </NavItem>
              <NavItem className='mx-4'>
                <NavLink to='/' onClick={logout}>
                  Logout
                </NavLink>
              </NavItem>
            </Nav>
          ) : (
            <Nav className='ms-auto'>
              <NavItem className='ms-4'>
                <NavLink to='/login'>Login</NavLink>
              </NavItem>
              <NavItem className='mx-4'>
                <NavLink to='/signup'>Sign Up</NavLink>
              </NavItem>
            </Nav>
          )}
        </Collapse>
      </Navbar>
    </div>
  );
}

export default NavBar;
