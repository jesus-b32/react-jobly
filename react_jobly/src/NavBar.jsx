import React, { useState } from 'react';
import { NavLink } from 'react-router-dom';
import {
  Navbar,
  Nav,
  NavItem,
  NavbarBrand,
  NavbarToggler,
  Collapse,
} from 'reactstrap';
import { Link } from 'react-router-dom';

function NavBar({ isLoggedIn }) {
  const [isOpen, setIsOpen] = useState(false);

  const toggle = () => setIsOpen(!isOpen);

  return (
    <div className='mb-4'>
      <Navbar color='dark' dark container='fluid' expand='md'>
        <NavbarBrand href='/'>Jobly</NavbarBrand>
        <NavbarToggler onClick={toggle} />
        <Collapse isOpen={isOpen} navbar>
          {isLoggedIn ? (
            <Nav className='me-auto' navbar>
              <NavItem>
                <Link to={'/companies'}>Companies</Link>
              </NavItem>
              <NavItem>
                <Link to={'/jobs'}>Jobs</Link>
              </NavItem>
              <NavItem>
                <Link to={'/profile'}>Profile</Link>
              </NavItem>
              <NavItem>
                <Link to={'/logout'}>Log Out</Link>
              </NavItem>
            </Nav>
          ) : (
            <Nav className='me-auto' navbar>
              <NavItem>
                <Link to={'/login'}>Login</Link>
              </NavItem>
              <NavItem>
                <Link to={'/signup'}>Sign Up</Link>
              </NavItem>
            </Nav>
          )}
        </Collapse>
      </Navbar>
    </div>
  );
}

export default NavBar;
