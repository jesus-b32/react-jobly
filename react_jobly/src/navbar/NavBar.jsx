import React, { useState, useContext } from 'react';
import UserContext from '../auth/UserContext';
import {
  Navbar,
  Nav,
  NavItem,
  NavbarBrand,
  NavbarToggler,
  Collapse,
} from 'reactstrap';
import { Link } from 'react-router-dom';

function NavBar({ logout }) {
  const { currentUser } = useContext(UserContext);
  const [isOpen, setIsOpen] = useState(false);

  const toggle = () => setIsOpen(!isOpen);

  return (
    <div className='mb-4'>
      <Navbar color='dark' dark container='fluid' expand='md'>
        <NavbarBrand href='/'>Jobly</NavbarBrand>
        <NavbarToggler onClick={toggle} />
        <Collapse isOpen={isOpen} navbar>
          {currentUser ? (
            <Nav className='me-auto' navbar>
              <NavItem>
                <Link to={'/companies'}>Companies</Link>
              </NavItem>
              <NavItem>
                <Link to={'/jobs'}>Jobs</Link>
              </NavItem>
              <NavItem>
                <Link to={'/profile'}>Profile</Link>
              </NavItem>
              <NavItem>
                <Link to={'/'} onClick={logout}>
                  Log Out {currentUser.firstName}
                </Link>
              </NavItem>
            </Nav>
          ) : (
            <Nav className='me-auto' navbar>
              <NavItem>
                <Link to={'/login'}>Login</Link>
              </NavItem>
              <NavItem>
                <Link to={'/signup'}>Sign Up</Link>
              </NavItem>
            </Nav>
          )}
        </Collapse>
      </Navbar>
    </div>
  );
}

export default NavBar;
