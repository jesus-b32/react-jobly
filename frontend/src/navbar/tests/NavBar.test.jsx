import { render } from '@testing-library/react';
import NavBar from '../NavBar';
import { MemoryRouter } from 'react-router-dom';
import { it, expect } from 'vitest';
import UserContext from '../../auth/UserContext';
import { testuser } from '../../tests/testCommon';

const TestNavBar = () => {
  const currentUser = testuser;
  return (
    <MemoryRouter>
      <UserContext.Provider value={{ currentUser }}>
        <NavBar />
      </UserContext.Provider>
    </MemoryRouter>
  );
};

it('should render without crashing', () => {
  render(<TestNavBar />);
});

it('should match snapshot', () => {
  const { asFragment } = render(<TestNavBar />);
  expect(asFragment()).toMatchSnapshot();
});
