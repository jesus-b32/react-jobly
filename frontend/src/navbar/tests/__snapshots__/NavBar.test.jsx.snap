// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="mb-4"
  >
    <nav
      class="navbar navbar-expand-md navbar-dark bg-dark"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          React-Jobly
        </a>
        <button
          aria-label="Toggle navigation"
          class="navbar-toggler"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse"
        >
          <ul
            class="ms-auto nav"
          >
            <li
              class="ms-4 nav-item"
            >
              <a
                class="text-decoration-none text-white bg-dark"
                href="/companies"
              >
                Companies
              </a>
            </li>
            <li
              class="ms-4 nav-item"
            >
              <a
                class="text-decoration-none text-white bg-dark"
                href="/jobs"
              >
                Jobs
              </a>
            </li>
            <li
              class="ms-4 nav-item"
            >
              <a
                class="text-decoration-none text-white bg-dark"
                href="/profile"
              >
                Profile
              </a>
            </li>
            <li
              class="mx-4 nav-item"
            >
              <a
                aria-current="page"
                class="text-decoration-none text-white bg-dark active"
                href="/"
              >
                Logout
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
